### PR TITLE
test: properly test web platform tests

### PR DESCRIPTION
**Pull request by anonrig**

### CODE REVIEW

**Comment by lemire** [comment]
It seems that the tests are not passing, right?

**Comment by anonrig** [comment]
> It seems that the tests are not passing, right?

Yes, we had a bug in the test runner, which is now fixed. I'm currently tackled with the failed test case.

File: src/unicode.cpp [hunk 1]
```diff
@@ -101,7 +101,8 @@ namespace ada::unicode {
    * @see https://github.com/nodejs/node/blob/main/src/node_url.cc#L226
    */
   std::string utf8_percent_encode(const std::string_view input, const uint8_t character_set[]) noexcept {
-    std::string result;
+    std::string result{};
+    result.reserve(input.length());
 
     for (auto iterator: input) {
       if (character_sets::BitAt(character_set, iterator)) {
```
Comment (lemire) [inline src/unicode.cpp:108]: To be consistent with the style, this should be `character_sets::bit_at`.

File: src/unicode.cpp [hunk 2]
```diff
@@ -101,7 +101,8 @@ namespace ada::unicode {
    * @see https://github.com/nodejs/node/blob/main/src/node_url.cc#L226
    */
   std::string utf8_percent_encode(const std::string_view input, const uint8_t character_set[]) noexcept {
-    std::string result;
+    std::string result{};
+    result.reserve(input.length());
 
     for (auto iterator: input) {
```
Comment (lemire) [inline src/unicode.cpp:107]: The 'auto' in the following loop...

```C++
for (auto iterator: input) {
```

is a `char`. Under many systems, a `char` is a signed type which is maybe not what you want. I recommend the following...

```C++
for (unsigned char iterator: input) {
```

Or a more explicit version...

```C++
for (uint8_t iterator: input) {
```

Comment (anonrig) [reply src/unicode.cpp]: Thanks I updated the implementation

**Pull request merged**